ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -1319,6 +1326,52 @@ void persistIndex() throws StoreException {
   }
 
   /**
+   * Re-validates the {@code token} to ensure that it is consistent with the current set of index segments. If it is
+   * not, a {@link StoreFindToken.Type#Uninitialized} token is returned.
+   * @param token the {@link StoreFindToken} to revalidate.
+   * @return {@code token} if is consistent with the current set of index segments, a new
+   * {@link StoreFindToken.Type#Uninitialized} token otherwise.
+   */
+  FindToken revalidateFindToken(FindToken token) {
```

Comment:
There is no revalidation required in `getBlobReadInfo()` (also it has no tokens).

`findDeletedEntriesSince()` seems to have been written with the assumption that it is used only internally in the store and its only use case is hard delete. It does not call `resetTokenIfRequired()` either. If we want to bring it up to par with `findEntriesSince()` (there is no reason to do that currently), we should do all the required things (out of scope for this change). Adding only the revalidate is not of much use.